firmware/src/hal/teensy/audio_teensy.cpp

```
@@ -0,0 +1,87 @@
+  return static_cast<int16_t>(clamped * kInt16FullScale);
+}
+
+// The engine as an audio library source: update() runs in the library's software
+// interrupt once per block and pulls one block through the app's callback.
+class EngineSource : public AudioStream {
+ public:
+  EngineSource() : AudioStream(0, nullptr) {}
+
+  void update() override {
+    if (callback_ == nullptr) return;
+    audio_block_t* left = allocate();
```

Clang (14.0.6)

[warning] variable 'left' is not initialized

(cppcoreguidelines-init-variables)